src/vs/workbench/contrib/terminal/browser/terminalService.ts

```diff
@@ -971,6 +981,29 @@ export class TerminalService implements ITerminalService {
 		return this._createTerminal(shellLaunchConfig, location, options);
 	}
 
+	async createDetachedXterm(options: IDetachedXTermOptions): Promise<IXtermTerminal> {
+		const ctor = await TerminalInstance.getXtermConstructor(this._keybindingService, this._contextKeyService);
+		const instance = this._instantiationService.createInstance(
+			XtermTerminal,
+			ctor,
+			this._configHelper,
+			options.cols,
+			options.rows,
+			{
+				getBackgroundColor: () => Color.transparent,
+			},
+			emptyTerminalCapabilityStore,
```

Comment:
One of the thing we want from the capability store is cwd detection as I think it's guaranteed to be the initial cwd for the test output. This is needed for links, but it currently depends on having a terminal instance so it would need a few more changes.

Would need to think more about the right solution, but it will probably work if you make `TerminalLinkContribution` take an optional instance and process manager, and then initialize it in `createDetachedXterm` like this:

https://github.com/microsoft/vscode/blob/b7fd221daf9c426b4f0ca126e70b3a1d58d15d91/src/vs/workbench/contrib/terminal/browser/terminalInstance.ts#L560-L580

Reply:
Still, we'd want the links even if we don't know the cwd. If we know it it just makes relative link detection better